### PR TITLE
Update 2: Add read/write and grant config info for external etcd in cluster guide

### LINKED ISSUE #2184: Sensu Go + etcd authentication (grant-permission)
As an engineer, 
I would like to ensure that my `backend-etcd` user has the correct permissions when `etcd` authentication is enabled.

## Feature Suggestion
Documentation on how to run Sensu Go while utilizing `etcd` authentication.  This will further enhance how to run Sensu Go securely.

## Possible Implementation
Add documentation to the follow page + h2
https://docs.sensu.io/sensu-go/latest/guides/securing-sensu/#etcd-user-role

## Context
Using etcd 3.4 with passwordless user account via mTLS with authentication enabled, I was unable to get Sensu Go `backend init` to run properly.

```
{"level":"warn","ts":"2020-02-13T15:31:24.552-0600","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-424eebe0-8294-4eb0-b146-7c456157de7c/REDACTED:2379","attempt":0,"error":"rpc error: code = PermissionDenied desc = etcdserver: permission denied"}
```

The concerning part is that I had scoped quite a bit into the user/role in etcd.

```
[root@REDACTED ~]# /opt/etcd/etcdctl user get REDACTED --detail
User: REDACTED

Role sensu_readwrite
KV Read:
	/
	[/, 0) (prefix /)
	[/sensu.io, /sensu.ip) (prefix /sensu.io)
	[/sensu.io/, /sensu.io0) (prefix /sensu.io/)
KV Write:
	/
	[/, 0) (prefix /)
	[/sensu.io, /sensu.ip) (prefix /sensu.io)
	[/sensu.io/, /sensu.io0) (prefix /sensu.io/)
```

As a test, I disabled etcd authentication and it was able to initialize the backend.  This means that this is definitely related to Sensu Go + permissions needed in etcd.  

I will continue working on getting them working in my environment and contribute as I can.

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Adds information about configuring read/write access and grant permission for etcd auth.
- Adds etcd compatibility info to datastore reference.

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2184

## Review Instructions
This PR replaces https://github.com/sensu/sensu-docs/pull/2792 to resolve conflicts.

See https://github.com/sensu/sensu-docs/pull/2792#discussion_r508552465 -- these instructions need to be tested to confirm they work. Cameron's comment indicates these instructions may not work, but it isn't clear how they should be corrected.

### CODE REVIEW

**Comment by hillaryfraley** [comment]
Eric is doing some testing: https://sensu.slack.com/archives/C60EEQFH8/p1619617443304000

**Comment by hillaryfraley** [comment]
@echlebek thought I should follow up -- I'm not sure if the etcd work for releases overlaps with the testing you had in mind for this PR. Any update?

**Comment by echlebek** [comment]
@hillaryfraley I'm sorry, I haven't had time to do this work. I will make an effort to get to it this week.

**Comment by hillaryfraley** [comment]
@echlebek just following up on this

File: content/sensu-go/6.2/operations/deploy-sensu/datastore.md [hunk 1]
```diff
@@ -496,3 +499,4 @@ enable_round_robin: true
 [18]: #datastore-specification
 [19]: ../install-sensu/#ports
 [20]: https://www.postgresql.org/docs/current/config-setting.html
+[21]: https://etcd.io/docs/v3.3.13/op-guide/clustering/
```
Comment (echlebek) [inline content/sensu-go/6.2/operations/deploy-sensu/datastore.md:503]: This link is out of date, we should update it to the 3.5 documentation. Also, I believe "clustering" is probably the wrong documentation section, there is likely another section on authorization.

Comment (echlebek) [reply content/sensu-go/6.2/operations/deploy-sensu/datastore.md:503]: Doc link: https://etcd.io/docs/v3.5/op-guide/authentication/

Comment (hillaryfraley) [reply content/sensu-go/6.2/operations/deploy-sensu/datastore.md:503]: Thanks @echlebek -- this PR is quite a bit behind, so those links are already fixed in the published content. I'll need to be careful of that kind of thing though when I finally do merge this, so thank you.

Is this the only thing you found? Or still testing?

Comment (echlebek) [reply content/sensu-go/6.2/operations/deploy-sensu/datastore.md:503]: I'm still testing... it's taking me a while to get the environment set up properly. Sorry this is taking so long.

Comment (hillaryfraley) [reply content/sensu-go/6.2/operations/deploy-sensu/datastore.md:503]: Not at all, I really appreciate your help!

**Comment by hillaryfraley** [comment]
@echlebek I know you're busy with releases, but I wondered if you have an update on this.

9/9/21: Eric is still working on this PR but is currently mainly focused on 6.5.0

**Comment by hillaryfraley** [comment]
@echlebek any update on this?

**Review by echlebek — APPROVED** [review]
Hillary, I haven't tested with a TLS-enabled etcd, but I believe these steps are correct. Sorry to take so long in approving this PR.

**Comment by hillaryfraley** [comment]
No problem @echlebek , thank you!

**Comment by hillaryfraley** [comment]
Closed in favor of https://github.com/sensu/sensu-docs/pull/3582

**Pull request closed without merging**